src/assembly/linux/main/README.md

```diff
@@ -32,6 +32,108 @@ Starting Linux demo... Use Ctrl+C to exit.
+
+```
+$ redo cosmos_config
+```
+
+This generates the plugin configuration file and its respective command and telemetry configurations in the `adamant_example/src/assembly/linux/build/cosmos/plugin/` directory.
+
+OpenC3 COSMOS runs inside of its own Docker containers. The COSMOS example project provides the necessary tools and commands to create an interface plugin from our generated configuration files. Navigate to the `cosmos-project` directory and run:
+
+```
+$ ./openc3.sh cli generate plugin LINUX_EXAMPLE
+```
+
+This creates a new directory called `openc3-cosmos-linux-example`. Navigate to this directory and run:
+
+```
+$ ./../openc3.sh cli generate target LINUX_EXAMPLE
+```
+
+This creates the plugin directory structure as well as template configuration files that our generated configuration will replace. These files are located at:
+
+```
+cosmos-project/openc3-cosmos-linux-example/plugin.txt
+cosmos-project/openc3-cosmos-linux-example/targets/LINUX_EXAMPLE/cmd_tlm/cmd.txt
+cosmos-project/openc3-cosmos-linux-example/targets/LINUX_EXAMPLE/cmd_tlm/tlm.txt
+```
+
+Replace the contents of these text files with the contents of the corresponding files generated by Redo, located in the Adamant example project directory at:
+
+```
+adamant_example/src/assembly/linux/build/cosmos/plugin
+```
+
+The COSMOS plugin uses a default CRC protocol provided by COSMOS, but also needs a custom protocol to apply Adamant's command packet checksum correctly. The protocol file is located in the Adamant directory at:
+
+```
+adamant/gnd/cosmos/cmd_checksum.rb
+```
+
+And must be placed in the COSMOS plugin directory at:
+
+```
+cosmos-project/openc3-cosmos-linux-example/targets/LINUX_EXAMPLE/lib/cmd_checksum.rb
+```
+
+A helper script, which takes the Adamant assembly name and COSMOS install directory as arguments, is provided to copy the plugin configuration files, and any custom protocols requested by the configuration, to the correct directories. If the COSMOS and Adamant example project directories are adjacent, complete the configuration by navigating to `adamant_example/gnd/cosmos` on the host machine and running:
+
+```
+$ ./install_cosmos_plugin.sh linux_example cosmos-project
+```
+
+The plugin can now be compiled. Navigate to the plugin directory at `cosmos-project/openc3-cosmos-linux-example` and run:
+
+```
+$ ./../openc3.sh cli rake build VERSION=1.0.0
+```
+
+Select `Click to select plugin .gem file to install` and navigate to the compiled plugin gem file at `cosmos-project/openc3-cosmos-linux-example/openc3-cosmos-linux-example-1.0.0.gem` to install our generated plugin. The plugin is templated to allow changing of parameters such as ports, but the default values are already set to correspond with this project.
+
+The plugin will be installed. Return to the Adamant Linux environment and start the Linux assembly by navigating to `adamant_example/src/assembly/linux/main` and running:
```

Comment:
Return to the Adamant virtual environment

Reply:
Again, instead of "navigating" just show:

```
$ cd adamant_example/src/assembly/linux/main
$ redo run
```

Reply:
Addressed in [6dbca33](https://github.com/lasp/adamant_example/pull/18/commits/6dbca3339b75ac5d0073265eb8555c826a575d77).